### PR TITLE
LibELF: Convert more string literals to StringView literals.

### PR DESCRIPTION
**Pull request by bgianfo**
Most of these won't have perf impact, but the optimization is
practically free, so no harm in fixing these up.

### CODE REVIEW

**Comment by alimpfard** [comment]
Before we go on an `operator""sv` spree, why not provide a `StringView(const char(&)[N])`?

**Comment by awesomekling** [comment]
Oh sure, that'd be nicer.

**Comment by bgianfo** [comment]
You can't because then you end up converting arrays which aren't necessarily filled to the size of the storage... so it's impossible to calculate length without strlen.
I already went down this hole :)

**Comment by bgianfo** [comment]
Take this code in the kernel for an example, I had an assert in my StringView constructor (what you suggested @alimpfard) that validate strlen vs size. 

https://github.com/SerenityOS/serenity/blob/7db8ccc0e47e09e6c5d015a042c374b43cdd43ed/Kernel/Net/LocalSocket.cpp#L151-L161

**Comment by alimpfard** [comment]
Hmmm...but an array ref isn't constexpr - maybe we can use that somehow...:thinking:
I'll take a little look down the hole too :)

**Comment by bgianfo** [comment]
> I'll take a little look down the hole too :)

I hope there is a way that I missed, but after staring at crashes for a while I realized what I was doing wasn't going to work. 
I'm sure you can prove me wrong 😁

**Comment by alimpfard** [comment]
Appears that it's not as straightforward as I thought, there's no real way to distinguish between the two as 'constexpr'-ness seems to disappear when you're inside a non-constexpr function (reasonable, maybe) - but if you're inside a constexpr function, looking at the array that was passed to you will blow up unconditionally if it were to be not a constant expression...

TL;DR, I guess we're going on an `operator""sv` spree.
